Wikipedia/Code/TalkPageCell.swift

```diff
@@ -117,37 +118,38 @@ final class TalkPageCell: UICollectionViewCell {
         topicView.configure(viewModel: viewModel)
         topicView.linkDelegate = linkDelegate
 
-        leadReplySpacer.isHidden = !viewModel.isThreadExpanded
-        leadReplyButton.isHidden = !viewModel.isThreadExpanded
-
-        let comments: [UIView] = stackView.arrangedSubviews.filter { view in view is TalkPageCellCommentView || view is TalkPageCellCommentSeparator }
-        stackView.arrangedSubviews.forEach { view in
-            if comments.contains(view) {
-                view.removeFromSuperview()
-            }
-        }
+        if viewModel.isThreadExpanded {
```

Comment:
I'm so glad you refactored my original approach here – I was never satisfied with it and it felt overly wasteful.